In [2]:
import pandas as pd

In [7]:

# Read in the data to be combined

crop_data = pd.read_csv('../data/crop_data.csv', index_col=False)
ndvi_data = pd.read_csv('../data/ndvi_data.csv', index_col=False)
gpm_data = pd.read_csv('../data/gpm_data.csv', index_col=False)


In [8]:
print("Number of rows:", len(crop_data))
print(crop_data.columns)

Number of rows: 6887
Index(['year', 'state_name', 'county_name', 'Value'], dtype='object')


By the end of this, we should have 6887 rows of observations where 1 row represents a single year of NDVI/GPM monthly data for a single county, with its corn production value as the response variable.

In [34]:
# Rename state_name and value columns to make a smoother joining process
crop_data.rename(columns={'state_name': 'state', 'Value': 'production'}, inplace=True)


crop_data['state'] = crop_data['state'].str.lower()
ndvi_data['state'] = ndvi_data['state'].str.lower()
gpm_data['state'] = gpm_data['state'].str.lower()

# Join the crop data on state_name and year with the ndvi data and the gpm data
df = pd.merge(crop_data, ndvi_data, on=['year', 'state'])
df = pd.merge(df, gpm_data, on=['year', 'state'])


In [37]:
df.head()

year     state                county_name  production  sample_val_April  \
0  2019  illinois  OTHER (COMBINED) COUNTIES  406.422491          0.299199   
1  2019  illinois                     BUREAU  957.531390          0.299199   
2  2019  illinois                    CARROLL  672.756230          0.299199   
3  2019  illinois                      HENRY  919.251471          0.299199   
4  2019  illinois                 JO DAVIESS  459.130408          0.299199   

   sample_val_May  sample_val_June  sample_val_July  sample_val_August  \
0        0.342944         0.440607         0.756683           0.859391   
1        0.342944         0.440607         0.756683           0.859391   
2        0.342944         0.440607         0.756683           0.859391   
3        0.342944         0.440607         0.756683           0.859391   
4        0.342944         0.440607         0.756683           0.859391   

   sample_val_September  ...  max_val_October  max_val_November  precip_april  \
0              0.709855  ...            0.452             0.307    141.291809   
1              0.709855  ...            0.452             0.307    141.291809   
2              0.709855  ...            0.452             0.307    141.291809   
3              0.709855  ...            0.452             0.307    141.291809   
4              0.709855  ...            0.452             0.307    141.291809   

   precip_may  precip_june  precip_july  precip_august  precip_september  \
0  195.180389   136.249619    84.797669      102.31855        131.399902   
1  195.180389   136.249619    84.797669      102.31855        131.399902   
2  195.180389   136.249619    84.797669      102.31855        131.399902   
3  195.180389   136.249619    84.797669      102.31855        131.399902   
4  195.180389   136.249619    84.797669      102.31855        131.399902   

   precip_october  precip_november  
0      122.606499        73.484634  
1      122.606499        73.484634  
2      122.606499        73.484634  
3      122.606499        73.484634  
4      122.606499        73.484634  

[5 rows x 52 columns]

In [38]:
# Some quick validation
crop_data.describe()

year    production
count  6887.000000   6887.000000
mean   2016.922898    387.619045
std       2.588095    446.986319
min    2013.000000      0.254014
25%    2015.000000    120.428064
50%    2017.000000    299.457172
75%    2019.000000    563.987411
max    2021.000000  19878.454274

In [39]:
gpm_data.describe()

year  precip_april  precip_may  precip_june  precip_july  \
count    90.000000     90.000000   90.000000    90.000000    90.000000   
mean   2017.000000     90.024325  121.514387   122.240312   102.391847   
std       2.596454     43.822501   40.707701    44.157856    33.627497   
min    2013.000000     23.207767   51.717552    44.304073    40.313267   
25%    2015.000000     56.931439   98.192106    94.201893    74.062828   
50%    2017.000000     85.180126  115.526626   118.468403    97.255829   
75%    2019.000000    121.953539  142.508893   149.150452   126.862953   
max    2021.000000    252.642654  262.954773   255.499573   210.203522   

       precip_august  precip_september  precip_october  precip_november  
count      90.000000         90.000000       90.000000        90.000000  
mean      100.283890         83.281575       80.764214        58.060511  
std        35.863768         36.698151       34.259680        36.995980  
min        27.387964         21.809034       11.636484         6.271477  
25%        75.914818         57.990151       53.640978        27.655852  
50%        97.767464         75.217903       81.919281        52.298391  
75%       126.136488        101.308161      105.301156        78.812794  
max       183.939621        199.307434      165.044006       224.119049

In [40]:
ndvi_data.describe()

year  sample_val_April  sample_val_May  sample_val_June  \
count   100.000000        100.000000      100.000000       100.000000   
mean   2017.500000          0.290115        0.339554         0.536783   
std       2.886751          0.037133        0.034531         0.073154   
min    2013.000000          0.232338        0.275682         0.405137   
25%    2015.000000          0.261735        0.315949         0.484361   
50%    2017.500000          0.284441        0.334982         0.529642   
75%    2020.000000          0.318594        0.361767         0.578193   
max    2022.000000          0.371543        0.430166         0.747198   

       sample_val_July  sample_val_August  sample_val_September  \
count       100.000000         100.000000            100.000000   
mean          0.809182           0.825528              0.572080   
std           0.053997           0.043852              0.064860   
min           0.643940           0.669246              0.428737   
25%           0.784724           0.812654              0.527642   
50%           0.824586           0.839373              0.568382   
75%           0.850085           0.853031              0.614535   
max           0.887348           0.882953              0.729046   

       sample_val_October  sample_val_November  sample_count_April  ...  \
count          100.000000           100.000000        1.000000e+02  ...   
mean             0.348015             0.287308        8.345864e+05  ...   
std              0.043739             0.031290        7.152060e+05  ...   
min              0.282752             0.241037        8.745100e+04  ...   
25%              0.317946             0.264302        2.890160e+05  ...   
50%              0.339653             0.280414        5.497505e+05  ...   
75%              0.363983             0.301226        1.432098e+06  ...   
max              0.494029             0.377816        2.709084e+06  ...   

       min_val_October  min_val_November  max_val_April  max_val_May  \
count       100.000000        100.000000     100.000000   100.000000   
mean          0.264340          0.245100       0.362520     0.462200   
std           0.026792          0.018315       0.038986     0.054844   
min           0.236000          0.218000       0.306000     0.403000   
25%           0.243000          0.228000       0.331000     0.428000   
50%           0.253000          0.245500       0.368000     0.456000   
75%           0.284000          0.258000       0.390000     0.466000   
max           0.325000          0.283000       0.492000     0.601000   

       max_val_June  max_val_July  max_val_August  max_val_September  \
count    100.000000    100.000000      100.000000         100.000000   
mean       0.794100      0.868300        0.870900           0.798800   
std        0.041149      0.031427        0.028613           0.048615   
min        0.732000      0.786000        0.797000           0.680000   
25%        0.775000      0.856000        0.860000           0.794000   
50%        0.785000      0.878000        0.877500           0.823000   
75%        0.828000      0.891000        0.890000           0.824000   
max        0.863000      0.897000        0.903000           0.843000   

       max_val_October  max_val_November  
count       100.000000        100.000000  
mean          0.452600          0.330440  
std           0.048636          0.034992  
min           0.386000          0.276000  
25%           0.432000          0.307000  
50%           0.445500          0.326000  
75%           0.455000          0.345000  
max           0.578000          0.418000  

[8 rows x 41 columns]

Awesome. We now have our combined data in a readable and understandable format. All that's left to do is to start modeling it.

In [41]:
df.to_csv('../data/dataset.csv', index=False)